File: TimeScope.Core/Entities/Project.cs
namespace TimeScope.Core.Entities;

public class Project : BaseEntity
{
    public string Name { get; set; } = string.Empty;
    public string? Description { get; set; }
    public Guid? GroupId { get; set; }
    public Group? Group { get; set; }

    // Relations
    public ICollection<Theme> Themes { get; set; } = new List<Theme>();
    public ICollection<WorkTask> Tasks { get; set; } = new List<WorkTask>();
}

File: TimeScope.Frontend/src/types/project.ts
export interface Project {
    id: string;
    name: string;
    description?: string;
    groupId?: string;
    createdAt: string;
    updatedAt?: string;
    isDeleted: boolean;
}

export interface Group {
    id: string;
    name: string;
    description?: string;
    createdAt: string;
    updatedAt?: string;
    isDeleted: boolean;
}

export interface Theme {
    id: string;
    name: string;
    color: string;
    description?: string;
    groupId?: string;
    projectId?: string;
    createdAt: string;
    updatedAt?: string;
    isDeleted: boolean;
}
